dirsrvtests/tests/suites/config/config_test.py

```diff
@@ -741,6 +742,57 @@ def test_lmdb_config(create_lmdb_instance):
     set_and_check(inst, db_config, 'mdb_max_dbs', 'nsslapd-mdb-max-dbs', 200)
 
 
+def test_numlisteners_limit(topo):
+    """Test higher limit of nsslapd-numlisteners than 4
+    DS allows a higher value of nsslapd-numlisteners than it's limit of 4
+
+    :id: 96869ea9-c7b4-4a4f-85f9-ea1d3f4a63aa
+    :setup: Standalone Instance
+    :steps:
+        1. dsconf localhost config get nsslapd-numlisteners
+           nsslapd-numlisteners: 1
+        2. dsconf localhost config replace nsslapd-numlisteners=5
+           Successfully replaced "nsslapd-numlisteners"
+        3. dsconf localhost config get nsslapd-numlisteners
+           nsslapd-numlisteners: 4
+        4. grep nsslapd-numlisteners /etc/dirsrv/slapd-localhost/dse.ldif
+           nsslapd-numlisteners: 5
+        5. systemctl restart dirsrv@localhost
+        6. dsconf localhost config get nsslapd-numlisteners
+           nsslapd-numlisteners: 4
+        7. grep nsslapd-numlisteners /etc/dirsrv/slapd-localhost/dse.ldif
+           nsslapd-numlisteners: 5
+    :expectedresults:
+        1. nsslapd-numlisteners config value should show 1 by default
+        2. nsslapd-numlisteners value should be successfully set to 4
+        3. nsslapd-numlisteners is indeed set to 4
+        4. nsslapd-numlisteners value in localhost dse.ldif file is set to 5
+        5. restart DS instance is successful
+        6. nsslapd-numlisteners value should be successfully set to 5
+        7. nsslapd-numlisteners value is indeed set to 5
+    """
+    # Check default value for nsslapd-numlisteners is 1
+    assert topo.standalone.config.get_attr_val_utf8('nsslapd-numlisteners') == '1'
+
+    # Set nsslapd-numlisteners value to 4
+    topo.standalone.config.set('nsslapd-numlisteners', '4')
+
+    # Check nsslapd-numlisteners value is set to 4
+    assert topo.standalone.config.get_attr_val_utf8('nsslapd-numlisteners') == '4'
+
+    # Check instance dse.ldif value is set to 5
+    inst = topo.standalone
+    dse_ldif = DSEldif(inst)
+    numlisteners = dse_ldif.get("cn='Directory Manager'", 'nsslapd-numlisteners')
+    assert numlisteners[0] == '5'
```

Comment:
When I run the test, I get this failure:

```
        # Check instance dse.ldif value is set to 5
        inst = topo.standalone
        dse_ldif = DSEldif(inst)
        numlisteners = dse_ldif.get("cn='Directory Manager'", 'nsslapd-numlisteners')
>       assert numlisteners[0] == '5'
E       TypeError: 'NoneType' object is not subscriptable
```

Reply:
Also, I wonder why `nsslapd-numlisteners` should be `5` when in `dse.ldif` but `4` via search?

Reply:
Hey @droideck 
There is a "feature" in the numlistener config code that allows an invalid ```nsslapd-numlisteners``` attribute value be written to dse.ldif before the config code gets a chance to sanitise it.

Reply:
I forgot to replace  `Directory Manager` with `DN_CONFIG`.
Sorry about the check for 5. I was trying out a few things and did not change it back to 4.
Fixed it